src/box/lua/upgrade.lua

```diff
@@ -1517,6 +1530,405 @@ local function upgrade()
     upgrade_from(get_version())
 end
 
+--------------------------------------------------------------------------------
+-- Downgrade part
+--------------------------------------------------------------------------------
+
+--------------------------------------------------------------------------------
+-- Tarantool 2.9.1
+--------------------------------------------------------------------------------
+
+local function restore_sql_builtin_functions(issue_handler)
+    if issue_handler.dry_run then
+        return
+    end
+    local sql_builtin_list = {
+        "TRIM", "TYPEOF", "PRINTF", "UNICODE", "CHAR", "HEX", "VERSION",
+        "QUOTE", "REPLACE", "SUBSTR", "GROUP_CONCAT", "JULIANDAY", "DATE",
+        "TIME", "DATETIME", "STRFTIME", "CURRENT_TIME", "CURRENT_TIMESTAMP",
+        "CURRENT_DATE", "LENGTH", "POSITION", "ROUND", "UPPER", "LOWER",
+        "IFNULL", "RANDOM", "CEIL", "CEILING", "CHARACTER_LENGTH",
+        "CHAR_LENGTH", "FLOOR", "MOD", "OCTET_LENGTH", "ROW_COUNT", "COUNT",
+        "LIKE", "ABS", "EXP", "LN", "POWER", "SQRT", "SUM", "TOTAL", "AVG",
+        "RANDOMBLOB", "NULLIF", "ZEROBLOB", "MIN", "MAX", "COALESCE", "EVERY",
+        "EXISTS", "EXTRACT", "SOME", "GREATER", "LESSER", "SOUNDEX",
+        "LIKELIHOOD", "LIKELY", "UNLIKELY", "_sql_stat_get", "_sql_stat_push",
+        "_sql_stat_init", "GREATEST", "LEAST",
+    }
+    local datetime = os.date("%Y-%m-%d %H:%M:%S")
+    local _func = box.space._func
+    -- Otherwise we can't insert SQL_BUILTIN function. It is prohibited
+    -- to add since 2.9.0.
+    set_system_triggers(false)
+    local status, err = pcall(function()
+        for _, func in ipairs(sql_builtin_list) do
+            if _func.index.name:get(func) == nil then
+                local t = _func:auto_increment{
+                    ADMIN, func, 1, 'SQL_BUILTIN', '', 'function', {}, 'any',
+                    'none', 'none', false, false, true, {}, setmap({}), '',
+                    datetime, datetime}
+                box.space._priv:replace{ADMIN, PUBLIC, 'function', t.id,
+                                        box.priv.X}
+            end
+        end
+    end)
+    if not status then
+        pcall(set_system_triggers, true)
```

Comment:
What if `pcall` fails? Seems strange and inconsistent.

Reply:
Then you can do much. Best effort. What is you suggestion?

Reply:
I guess we should panic. Because AFAIU from @alyapunov's comment, this is not acceptable:
> If we fail to insert, the system triggers will be disabled forever.

Reply:
`set_system_triggers` should not fail normally (except for OOM) so decided we don't need to panic or using pcall.